### PR TITLE
Fix gas used and fee

### PR DESCRIPTION
**Pull request by miiu96**
## Reasoning behind the pull request
- Bug-fix the problem of how is calculated the `gasUsed` and `fee` fields for the transaction structure that is returned by the `/transction/${txHash}` endpoint and for the `outport` package
  
## Proposed changes
- In case of an ESDT Transfer ( fungible token transfer, nft transfer, or multi transfer) with smart contract calls the fee field will be set with initialPaidFee and the gasUsed field will be set with gasLimit of the transaction. This solution will solve the case when we have this kind of transaction and it fails (in that case these fields were incorrect).

## Testing procedure
- For a failed transaction with an ESDT transfer and SC call check if the `gasUsed` and `fee` fields are calculated correctly.

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: node/external/transactionAPI/gasUsedAndFeeProcessor.go [hunk 1]
```diff
@@ -74,3 +76,33 @@ func (gfp *gasUsedAndFeeProcessor) setGasUsedAndFeeBaseOnRefundValue(tx *transac
 	tx.GasUsed = gasUsed
 	tx.Fee = fee.String()
 }
+
+func (gfp *gasUsedAndFeeProcessor) isESDTOperationWithSCCall(tx *transaction.ApiTransactionResult) bool {
+	isESDTTransferOperation := tx.Operation == core.BuiltInFunctionESDTTransfer ||
+		tx.Operation == core.BuiltInFunctionESDTNFTTransfer || tx.Operation == core.BuiltInFunctionMultiESDTNFTTransfer
+
+	receiverIsSC := core.IsSmartContractAddress(tx.Tx.GetRcvAddr())
```
Comment (bogdan-rosianu) [inline node/external/transactionAPI/gasUsedAndFeeProcessor.go:84]: isReceiverSC

File: outport/process/transactionsfee/transactionChecker.go [hunk 2]
```diff
@@ -12,6 +12,31 @@ import (
 	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
 )
 
+func (tep *transactionsFeeProcessor) isESDTOperationWithSCCall(tx data.TransactionHandlerWithGasUsedAndFee) bool {
+	res := tep.dataFieldParser.Parse(tx.GetData(), tx.GetSndAddr(), tx.GetRcvAddr(), tep.shardCoordinator.NumberOfShards())
+
+	isESDTTransferOperation := res.Operation == core.BuiltInFunctionESDTTransfer ||
+		res.Operation == core.BuiltInFunctionESDTNFTTransfer || res.Operation == core.BuiltInFunctionMultiESDTNFTTransfer
+
+	receiverIsSC := core.IsSmartContractAddress(tx.GetRcvAddr())
```
Comment (bogdan-rosianu) [inline outport/process/transactionsfee/transactionChecker.go:21]: isReceiverSC

File: node/external/transactionAPI/gasUsedAndFeeProcessor.go [hunk 3]
```diff
@@ -74,3 +76,33 @@ func (gfp *gasUsedAndFeeProcessor) setGasUsedAndFeeBaseOnRefundValue(tx *transac
...
+	}
+
+	if len(tx.Receivers) == 0 {
+		return false
+	}
+
+	receiver := tx.Receivers[0]
+	decodedReceiver, err := gfp.pubKeyConverter.Decode(receiver)
+	if err != nil {
+		log.Warn("gpf.isESDTOperationWithSCCall cannot decode receiver address", "error", err.Error())
```
[the remark this answers is not in the document]
Comment (miiu96) [reply node/external/transactionAPI/gasUsedAndFeeProcessor.go:101]: done

File: node/external/transactionAPI/gasUsedAndFeeProcessor.go (hunk 1 above)
Comment (miiu96) [reply node/external/transactionAPI/gasUsedAndFeeProcessor.go]: done

File: outport/process/transactionsfee/transactionChecker.go (hunk 2 above)
Comment (miiu96) [reply outport/process/transactionsfee/transactionChecker.go]: done

**Review by iulianpascalau — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed

**Review by gabi-vuls — COMMENTED** [review]
@@                            Log scanner                                  @@
##                            fix-gas-used-and-fee-api-tx-and-outport      ##
================================================================================
- Known Warnings              9
- New Warnings                2
- Known Errors                0
- New Errors                  0
- Panics                      0
================================================================================
- block hash does not match                  2422
- miniblocks does not match                  0
- miniblock hash does not match              0
- receipts hash missmatch                    0
- wrong nonce in block                       1051
================================================================================
-  No jailed nodes on the thestnet
================================================================================

**Pull request merged**